lib/Doctrine/ORM/Mapping/JoinColumnData.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ORM\Mapping;
+
+use ArrayAccess;
+
+use function property_exists;
+
+/** @template-implements ArrayAccess<string, mixed> */
+final class JoinColumnData implements ArrayAccess
```

Comment:
Data is very generic and doesn't seem like a good fit. `JoinColumnMapping`? `JoinColumnDefinition`? `JoinColumn`?

Reply:
`JoinColumnMapping` 👍